### PR TITLE
refactor: cleanups & fixes

### PR DESCRIPTION
**Pull request by xMooddyy**
This PR removes a couple of duplicate code & redundant stuff, and makes some changes to errors & README.

### CODE REVIEW

**Comment by xMooddyy** [comment]
Just pushed a fix to the following error:

Whenever creating a modal button/select menu, if the user opens the modal and closes it, the whole interaction fails and they cannot click any button anymore. 

The fix:

Revert back to using `.createMessageComponentCollector`, this allows the user to still click buttons even after opening and closing a modal.

File: examples/simple-marketplace/states/InitialState.ts [hunk 1]
```diff
@@ -4,38 +4,37 @@ import { MarketplacePostContext, posts } from '..';
...
+	message: () => ({
+		ephemeral: true,
+		embeds: [],
+		components: [
+			[
+				{
+					type: PromptComponentType.Button,
+					component: () => new ButtonBuilder().setLabel('Create new Post').setStyle(ButtonStyle.Primary),
+					callback: (ctx) => {
+						ctx.previousState = [initialState.name];
```
Comment (sm3136) [inline examples/simple-marketplace/states/InitialState.ts:18]: Handled by the prompt system on its own, shouldn't need to do this anymore.

File: examples/simple-marketplace/states/preCreatePostState.ts [hunk 2]
```diff
@@ -4,30 +4,33 @@ import { MarketplacePostContext } from '..';
...
+						new StringSelectMenuBuilder().setPlaceholder('* Select post type').addOptions([
+							{ label: 'Hiring', value: 'Hiring', default: ctx.selectedPost?.postType === 'Hiring' },
+							{
+								label: 'Hireable',
+								value: 'Hireable',
+								default: ctx.selectedPost?.postType === 'Hireable',
+							},
+						]),
+					callback: (ctx) => {
+						ctx.previousState.push(preCreatePostState.name);
```
Comment (sm3136) [inline examples/simple-marketplace/states/preCreatePostState.ts:26]: Handled by the prompt system on its own, shouldn't need to do this anymore.

**Review by sm3136 — CHANGES_REQUESTED** [review]
I would try to make some changes that revolve around those comments. I think one of them has a potential to improve speed, others are kind of just nit picks.

**Pull request merged**